common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/Indexer.scala

```diff
@@ -25,14 +24,53 @@ trait Indexable[T] {
   def version(document: T): Int
 }
 
+/**
+  * Versioning:
+  *
+  * When the merger makes the decision to merge some works, it modifies
+  * the content of the affected works. Despite the content of these works
+  * being modified, their version remains the same. However, the merger
+  * attaches the number of merged sources (`nMergedSources`) to the target
+  * work and may choose to redirect some of the source works.
+  *
+  * When we ingest those works, we need to make sure that the merger
+  * modified works are never overridden by unmerged works for the same
+  * version still running through the pipeline.
+  *
+  * We also need to make sure that, if a work is modified by a source in
+  * such a way that it shouldn't be merged (or redirected) anymore, the
+  * new unmerged version is ingested and never replaced by the previous
+  * merged version still running through the pipeline.
+  *
+  * We can do that by ingesting works into Elasticsearch with a document
+  * version derived from a combination of work version, number of merged
+  * sources and work type.
+  *
+  * More specifically, by multiplying the work version by 10, we make sure
+  * that a new version of a work always wins over previous versions
+  * (merged or unmerged, provided that there are fewer than 10 merged
+  * sources, which we think is a reasonable assumption).
+  *
+  * We make sure that a merger modified work always wins over other works
+  * with the same version, by adding the number of merged sources to
+  * work.version * 10.
+  *
+  * The exact same logic is applied to image versions, using both the
+  * merge state and the version numbers of the image's SourceWorks.
+  */
 object Indexable {
 
   implicit val imageIndexable: Indexable[AugmentedImage] =
     new Indexable[AugmentedImage] {
       def id(image: AugmentedImage) =
         image.id.canonicalId
+
       def version(image: AugmentedImage) =
-        image.version
+        10 * (image.version + image.source.version) + (image.source match {
```

Comment:
What source version id this one? The calculated one for es or the one from the transformer?

Reply:
From the transformer